src/iop/demosaic.c

```diff
@@ -2480,6 +2485,47 @@ static void passthrough_monochrome(float *out, const float *const in, dt_iop_roi
   }
 }
 
+static void blk4_monochrome(float *out, const float *const in, dt_iop_roi_t *const roi_out,
+                                   const dt_iop_roi_t *const roi_in, const uint32_t filters, const uint32_t rgb_filter)
+{
+  assert(roi_in->width >= roi_out->width);
+  assert(roi_in->height >= roi_out->height);
+
+  const float rgb_pars[4][4] = {
+    { 0.2225f, 0.7169f / 2.0f, 0.0606f, 0.0f }, // luminance
+    { 0.5f, 0.5f / 2.0f, 0.0f, 0.0f },          // yellow
+    { 1.0f, 0.0f, 0.0f, 0.0f },                 // red
+    { 0.3333f, 0.3333f / 2.0f, 0.3333f, 0.0f } // equal
+  };
+
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(in, roi_out, roi_in) \
+  shared(out, rgb_pars, filters,rgb_filter) \
+  schedule(static)
+#endif
+
+  for(int j = 0; j < (roi_out->height); j++)
+  {
+    for(int i = 0; i < (roi_out->width); i++)
+    {
+      const int lim_x = MIN(i + 1 + roi_out->x, roi_in->width-1);
+      const int lim_y = MIN(j + 1 + roi_out->y, roi_in->height-1);
+      const int src_x = MIN(i + roi_out->x, roi_in->width-1);
+      const int src_y = MIN(j + roi_out->y, roi_in->height-1);
+
+      const float data =
+        in[(size_t)((size_t)src_y) * roi_in->width + src_x] * rgb_pars[rgb_filter][FC(src_y, src_x, filters)] +
+        in[(size_t)((size_t)src_y) * roi_in->width + lim_x] * rgb_pars[rgb_filter][FC(src_y, lim_x, filters)] +
+        in[(size_t)((size_t)lim_y) * roi_in->width + src_x] * rgb_pars[rgb_filter][FC(lim_y, src_x, filters)] +
+        in[(size_t)((size_t)lim_y) * roi_in->width + lim_x] * rgb_pars[rgb_filter][FC(lim_y, lim_x, filters)];
+
```

Comment:
so this is a simple local arithmetic mean with RGB weighting ? Isn't that blurry ?

Reply:
It is indeed. Simple weighed mean over a 2x2 sensor area vs clever algo using a larger area. I think the point is the algo is optimized for colour 'regenerating'.

Anyway i tested quite a lot and was surprised how well this works on a pixel-peeping level.

Reply:
did you try with horizontal/vertical/diagonal sharp details (architecture photos are good candidates for that) ? The other end of the demosaicing issues, if it's not blurry, is staircasing effects/aliasing. The clever algos try to mitigate both ends of the issue.

Reply:
Yes i tried and especially for straight borders amaze is less blurry by ~1 rgb-pixel. 
![Bildschirmfoto von 2020-04-26 06-27-22](https://user-images.githubusercontent.com/50982232/80297942-5a639880-8788-11ea-865a-6fba6a93b156.png)
![Bildschirmfoto von 2020-04-26 06-26-53](https://user-images.githubusercontent.com/50982232/80297947-6baca500-8788-11ea-9b66-c0da3d08b422.png)

Reply:
WOW that's amazing.